code/datums/components/bloodysoles.dm

```diff
@@ -44,7 +44,20 @@
  * Returns true if the parent item is obscured by something else that the wielder is wearing
  */
 /datum/component/bloodysoles/proc/is_obscured()
-	return wielder.check_obscured_slots(TRUE) & equipped_slot
+	return wielder.check_obscured_slots(TRUE) & equipped_slot || is_under_feet_covered()
+
+/**
+ * Returns true if the parent item is worn in the ITEM_SLOT_ICLOTHING slot and the
+ * wielder is wearing something on their shoes.
+ *
+ * Allows for jumpsuits to cover feet without getting all bloodied when their wearer
+ * is wearing shoes.
+ */
+/datum/component/bloodysoles/proc/is_under_feet_covered()
+	if(equipped_slot != ITEM_SLOT_ICLOTHING)
```

Comment:
Should be using a bitwise operation?

Reply:
I first thought about that, but then I remembered that an item should only really be equipped to a singular slot, and I don't know if we have anything at all in the code that does otherwise.

Reply:
I know we went through a while ago and made all the slot checks bitwise operations even though they were only checking for a single slot most of the time, so it's good for consistency at least